### PR TITLE
Disable decorators via config

### LINKED ISSUE #1143: Add the posibility to customize default decorators
Hi! 

We are running into trouble with dd-trace-ot ... we don't want the protection against 404 spam because we do make that protection somewhere else and dd-trace implementation messes up our traces. Would you be open to adding the option to customize default decorators? 

I'm ok with making the PR. Thanks!

### PR DESCRIPTION
**Pull request by randomanderson**
Adds the ability to disable individual decorators with config.

Closes #1143

### CODE REVIEW

File: dd-trace-api/src/main/java/datadog/trace/api/Config.java [hunk 1]
```diff
@@ -596,6 +596,10 @@ public boolean isJmxFetchIntegrationEnabled(
     return jmxFetchIntegrationEnabled(integrationNames, defaultEnabled);
   }
 
+  public boolean isDecoratorEnabled(final String name) {
+    return getBooleanSettingFromEnvironment("trace." + name.toLowerCase() + ".enabled", true);
```
Comment (tylerbenson) [inline dd-trace-api/src/main/java/datadog/trace/api/Config.java]: @andrewsouthard1 any objections to this config setting?

File: dd-trace-ot/src/main/java/datadog/opentracing/decorators/DDDecoratorsFactory.java [hunk 2]
```diff
@@ -4,31 +4,44 @@
...
+            new ResourceNameDecorator(),
+            new ServiceNameDecorator(),
+            new ServiceNameDecorator("service", false),
+            new ServletContextDecorator(),
+            new SpanTypeDecorator(),
+            new Status404Decorator(),
+            new Status5XXDecorator(),
+            new URLAsResourceName())) {
+
+      if (Config.get().isDecoratorEnabled(decorator.getClass().getSimpleName())) {
```
Comment (dougqh) [inline dd-trace-ot/src/main/java/datadog/opentracing/decorators/DDDecoratorsFactory.java]: I realize I'm a bit late, but none-the-less...
I think it is better to have a separate name other than the class name for any configuration setting.

Fortunately, It won't be hard for us to make that change in the future.

File: dd-trace-api/src/main/java/datadog/trace/api/Config.java (hunk 1 above)
Comment (andrewsouthard1) [reply dd-trace-api/src/main/java/datadog/trace/api/Config.java]: 👍

**Comment by tylerbenson** [comment]
FYI, system properties are case sensitive, so the property needs to be set as lowercase.  I'll update the PR description.

**Pull request merged**